### PR TITLE
Support for storing images in RAM (Please review PR #11 first)

### PR DESCRIPTION
**Pull request by j-simpson1**
- to test set **cache_in_memory=True** in train_dataset
- note: it crashed my laptop with 16GB of RAM, so not sure it is completely practical unless you've got lots of RAM.

### CODE REVIEW

**Review by timfarkas — APPROVED** [review]

**Review by timfarkas — COMMENTED** [review]
awesome testing now
